module/spl/spl-taskq.c

```diff
@@ -366,9 +368,13 @@ taskq_find(taskq_t *tq, taskqid_t id, int *active)
 	list_for_each(l, &tq->tq_active_list) {
 		tqt = list_entry(l, taskq_thread_t, tqt_active_list);
 		if (tqt->tqt_id == id) {
-			t = tqt->tqt_task;
 			*active = 1;
-			return (t);
+			/*
+			 * Instead of returning tqt_task, we just return a non
+			 * NULL value to prevent misuse, since tqt_task only
+			 * have two valid fields.
+			 */
+			return ((taskq_ent_t *)-1L);
```

Comment:
If we can't safely return the `taskq_ent_t *` in the active case, then we don't we do something like return `ERR_PTR(-EBUSY)` and remove the `active` argument as a simplification.  There are only two callers which would need to be updated.